virtcontainers/kata_agent.go

```diff
@@ -761,6 +761,10 @@ func (k *kataAgent) getDNS(sandbox *Sandbox) ([]string, error) {
 func (k *kataAgent) startSandbox(sandbox *Sandbox) error {
 	span, _ := k.trace("startSandbox")
 	defer span.Finish()
+	ociSpec := sandbox.GetPatchedOCISpec()
```

Comment:
I think this may not work. `k.startSandbox` is called after `h.startSandbox`. I still think that this should be in `startVM()`

https://github.com/kata-containers/runtime/blob/83eef430bd9eaa3d313bb3c1e73bdcbe4fbf23cc/virtcontainers/sandbox.go#L1016-L1050